cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -723,6 +723,7 @@ pub(crate) fn define(
     e.enc_both(band.bind(B1), rec_rr.opcodes(&AND));
     e.enc_both(bor.bind(B1), rec_rr.opcodes(&OR));
     e.enc_both(bxor.bind(B1), rec_rr.opcodes(&XOR));
+    e.enc_both(bnot.bind(B1), rec_ur.opcodes(&NOT).rrr(2));
```

Comment:
x86 `NOT` definition: "Performs a bitwise NOT operation (each 1 is set to 0, and each 0 is set to 1) on the destination operand and stores the result in the destination operand location. The destination operand can be a register or a memory location." 

If we `NOT` something like `...000001` we will get `...111110`. @bnjbvr, we're sure Cranelift will prevent subsequent operations from accessing these newly-set bits?

Reply:
No. `brz` requires the full byte to be zero, not just the lsb: https://github.com/ash-tamraz/cranelift/blob/5ac23d2845c971b5fef5583ad2b05d57affc9878/cranelift-codegen/meta/src/isa/x86/encodings.rs#L1346

Reply:
Then we might need a different encoding; how about the following:
 - `BTC %reg, 0`: Selects the bit in a bit string (specified with the first operand, called the bit base) at the bit-position designated by the bit offset operand (second operand), stores the value of the bit in the CF flag, and complements the selected bit in the bit string.

Reply:
> complements the selected bit in the bit string.

That would mean an extra copie is required when using a bool after a brz/brnz.

Reply:
Hm, here's a less destructive version: `BT  + SETNC` (store selected bit in the CF flag + set byte if not carry (CF=0)). But it is two instructions...

Reply:
`bt` + `jc`/`jnc` for implementing `brz` and `brnz` is enough right?

Reply:
Duh, right; that's probably why it wasn't implemented in the first place. We could probably add a check in the verifier that any `b1` value only contains a 0 or 1.

So just thinking a bit about this: I wonder if we could use Bool's algebra and replace `bnot.b1 x` with `bxor_imm.b1 x, 1`. This is a bit harder to implement, though...